src/core/dataset.py

```
@@ -38,30 +42,52 @@ def __getitem__(self, index):
                 input_ids = self.tokenizer.encode(system_text, add_special_tokens=False)
                 target_mask = [0] * len(input_ids)
 
+        # setting tool information
+        if "tools" in data.keys() and data["tools"]:
```

Ruff

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)